lib/PHPGGC/Enhancement/PublicAttributes.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace PHPGGC\Enhancement;
+
+/**
+* Public Attributes
+* Attempts to convert references to protected properties within the serialized
+* payload to public.
+*
+* This can be useful because when PHP serializes a protected property of an
+* object it prepends the property name with an asterisk surrounded by null
+* bytes, which are easy to lose if the payload is transmitted or stored as plain
+* text without encoding. If that happens, the payload will fail to unserialize
+* because the string length of the property name (and the name itself) will be
+* incorrect.
+*
+* As an added bonus, payloads are slightly smaller without the prefixes.
+*
+* Converting protected properties to public tends to work in more recent PHP
+* versions but can cause problems in older versions such as PHP 5.6.
+*/
+class PublicAttributes extends Enhancement
+{
+
+    /**
+     * Post process step of the public-attributes technique: removes prefixes
+     * denoting protected properties, converting them to public.
+     */
+    public function process_serialized($serialized)
+    {
+       $encoded_payload = urlencode($serialized);
```

Comment:
Yup, otherwise you get something like `PHP Warning:  preg_match(): Null byte in regex`.

The encoding generally makes it easier to see what's going on (within reason!).

I'll add a comment.